components/bootloader_support/src/esp_image_format.c

```diff
@@ -125,17 +126,26 @@ static esp_err_t image_load(esp_image_load_mode_t mode, const esp_partition_pos_
         goto err;
     }
 
-    // Calculate SHA-256 of image if secure boot is on, or if image has a hash appended
+#ifdef BOOTLOADER_BUILD
+    if (mode != ESP_IMAGE_LOAD_NO_VALIDATE)
+#else
+    if (1)
+#endif
```

Comment:
To simplify the code here and also for a more readable version of the `if(NULL != checksum)` check below, how about setting up a `bool do_verify` at the top of the function, which is true for non-bootloader builds and true for bootloader builds if `mode != ESP_IMAGE_LOAD_NO_VALIDATE`?